app/services/BackgroundTaskService.js

```diff
@@ -0,0 +1,135 @@
+import BackgroundFetch from 'react-native-background-fetch';
+import PushNotification from 'react-native-push-notification';
+import { IntersectSet } from '../helpers/Intersect';
+import { GetStoreData, SetStoreData } from '../helpers/General';
+import { isPlatformiOS } from './../Util';
+import languages from '../locales/languages';
+
+const INTERSECT_INTERVAL = 30 / 60; // In Minutes
+
+export function executeTask() {
+  check_intersect = () => {
+    // This function is called once every 12 hours.  It should do several things:
+
+    console.log(
+      'Intersect tick entering on',
+      isPlatformiOS() ? 'iOS' : 'Android',
+    );
+    // this.findNewAuthorities(); NOT IMPLEMENTED YET
+
+    // Get the user's health authorities
+    GetStoreData('AUTHORITY_SOURCE_SETTINGS')
+      .then(authority_list => {
+        if (!authority_list) {
+          // DEBUG: Force a test list
+          // authority_list = [
+          //  {
+          //    name: 'Platte County Health',
+          //    url:
+          //      'https://raw.githack.com/tripleblindmarket/safe-places/develop/examples/safe-paths.json',
+          //  },
+          //];
+          console.log('No authorities', authority_list);
+          return;
+        }
+
+        let name_news = [];
+        if (authority_list) {
+          // Pull down data from all the registered health authorities
+          authority_list = JSON.parse(authority_list);
+          for (const authority of authority_list) {
+            console.log('[auth]', authority);
+            fetch(authority.url)
+              .then(response => response.json())
+              .then(responseJson => {
+                // Example response =
+                // { "authority_name":  "Steve's Fake Testing Organization",
+                //   "publish_date_utc": "1584924583",
+                //   "info_website": "https://www.who.int/emergencies/diseases/novel-coronavirus-2019",
+                //   "concern_points":
+                //    [
+                //      { "time": 123, "latitude": 12.34, "longitude": 12.34},
+                //      { "time": 456, "latitude": 12.34, "longitude": 12.34}
+                //    ]
+                // }
+
+                // Update cache of info about the authority
+                // TODO: Add an "info_newsflash" UTC timestamp and popup a
+                //       notification if that changes, i.e. if there is a newsflash?
+                name_news.push({
+                  name: responseJson.authority_name,
+                  news_url: responseJson.info_website,
+                });
+
+                // TODO: Look at "publish_date_utc".  We should notify users if
+                //       their authority is no longer functioning.)
+
+                IntersectSet(responseJson.concern_points, dayBin => {
```

Comment:
Is there an issue here with calls that can happen via the background service happening concurrently with the UI?  Seems possible that the user could have the background service chugging away at the Intersect, while the user opens the app and the UI thread _also_ kicks off doing it.  Might end up with a race condition and the user getting notified twice in succession.  I guess it all depends upon how long the `IntersectSet` takes.